agent/stats/queue.go

```diff
@@ -186,6 +190,70 @@ func (queue *Queue) GetStorageStatsSet() (*ecstcs.StorageStatsSet, error) {
 	return storageStatsSet, errOut
 }
 
+// GetRestartStatsSet gets the stats set for container restarts
+func (queue *Queue) GetRestartStatsSet() (*ecstcs.RestartStatsSet, error) {
+	return queue.getRestartStatsSet(getRestartCount)
+}
+
+func (queue *Queue) getRestartStatsSet(getInt getIntPointerFunc) (*ecstcs.RestartStatsSet, error) {
+	queue.lock.Lock()
+	defer queue.lock.Unlock()
+
+	var firstStat, lastStat int64
+	firstStat = -1
+
+	queueLength := len(queue.buffer)
+	if queueLength < 2 {
+		// Need at least 2 data points to calculate this.
+		return nil, fmt.Errorf("need at least 2 data points in queue to calculate restart stats set")
```

Comment:
Are 2 data points strictly required to send container restart metric to TACS? Why consider a queue containing a single stat with a restart count as invalid/an error case?

Reply:
this is because the restart count metric is a monotonic counter that is always increasing, so what we send is the diff between two metrics.

If we accepted a single metric, then we might send a huge metric after an agent/instance reboot, for example if our stats queue looked like this before/after a reboot:

```
[0, 10, 13, 21]
(reboot happens)
[21]
```

So in this case we would not want to send a metric of "21" on startup.

Reply:
Got it, thanks for the example and explanation!